src/common/spinctrlcmn.cpp

```diff
@@ -137,4 +137,10 @@ wxSize wxSpinCtrlImpl::GetBestSize(const wxControl* spin,
     return spin->GetSizeFromText(largestString);
 }
 
+bool wxSpinCtrlImpl::AreParametersValid(int minVal, int maxVal, int base)
+{
+    // Negative values in the range are allowed only if base == 10
+    return base == 10 || (minVal >= 0 && maxVal >= 0);
```

Comment:
Maybe the test would be clearer if it went something in line of:
`return base == 10 || (maxVal >= minVal && minVal >= 0);`

Reply:
> Out of curiosity, is having `maxVal < minVal` allowed?

Yes, native up-down control under MSW supports this inverted boundaries (see [UDM_SETRANGE docs](https://docs.microsoft.com/en-us/windows/win32/controls/udm-setrange) and wxSpinCtrl under wxMSW supports this too - it's seen e.g. in the wxSpinCtrl::SetRange() code.
Native GtkSpinButton doesn't seem to support this inversion (widget just locks at maxVal) but wxSpinCtrl under wxGTK doesn't explicitly prevent from passing maxVal < minVal. The same for generic wxSpinCtrl.

Reply:
Thanks, I remembered that MSW allowed this, but I thought we didn't because of portability considerations. Apparently it was just another thing that we planned/discussed but never did however... It's not worth disallowing it if this works under MSW currently, but we probably should reorder min/max under the other platforms to allow the same code result in something usable (even if not behaving identically) under all of them, what do you think?

This is not directly related to this PR, of course, which can be merged right now, unless you plan on renaming the function as suggested (or I could do it myself when merging it if you prefer).

Reply:
I will merge this PR once I rename this function and add some unit tests.